### PR TITLE
Impersonate UI

### LINKED ISSUE #4241: 2.0 Settings / Admin {View}
https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=18408%3A171954

### LINKED ISSUE #4270: /settings - admin panel is visible for non admins
I tried impersonating my test account I made but it says I don’t have permission which is good but it should be hidden

### PR DESCRIPTION
**Pull request by sean-brydon**
Adds impersonate UI to settings

Also adds a new permission container 

CleanShot 2022-09-07 at 15 50 51

### CODE REVIEW

File: packages/ui/v2/core/PermissionContainer.tsx
Comment (PeerRich) [inline packages/ui/v2/core/PermissionContainer.tsx:30]: this is cool. bascially hides content for non-admins?

**Review by PeerRich — APPROVED** [review]
great stuff. love that new admin check component

**Comment by sean-brydon** [comment]
@zomars could you sanity check this admin component before we merge?

File: packages/ui/v2/core/PermissionContainer.tsx
Comment (zomars) [inline packages/ui/v2/core/PermissionContainer.tsx:9]: I'm not 100% convinced with the skip approach but it's ok for now

File: packages/ui/v2/core/PermissionContainer.tsx
Comment (zomars) [inline packages/ui/v2/core/PermissionContainer.tsx:15]: Let move this file to `@calcom/features/permissions`

File: packages/ui/v2/core/navigation/tabs/HorizontalTabItem.tsx
Comment (zomars) [inline packages/ui/v2/core/navigation/tabs/HorizontalTabItem.tsx:9]: HorizontalTabs should not contain outside logic. We can either create a container with logic or figure another way to achieve this

File: packages/ui/v2/core/navigation/tabs/HorizontalTabItem.tsx
Comment (zomars) [inline packages/ui/v2/core/navigation/tabs/HorizontalTabItem.tsx:14]: If rather filter out in a parent container than adding logic to a dumb component

File: packages/ui/v2/core/navigation/tabs/VerticalTabItem.tsx
Comment (zomars) [inline packages/ui/v2/core/navigation/tabs/VerticalTabItem.tsx:22]: Same here

File: packages/ui/v2/core/navigation/tabs/HorizontalTabItem.tsx
Comment (PeerRich) [reply packages/ui/v2/core/navigation/tabs/HorizontalTabItem.tsx:14]: true. i didnt see this

File: packages/ui/v2/core/PermissionContainer.tsx
Comment (zomars) [reply packages/ui/v2/core/PermissionContainer.tsx:9]: From Twist:

> Also. Instead of the adminRequired prop. You can have an `const adminRequiredKeys = [“impersonation”]` and check against that array

**Review by zomars — CHANGES_REQUESTED** [review]
Blocking until feedback is addressed

**Comment by PeerRich** [comment]
related: https://github.com/calcom/cal.com/issues/4270

it seems like all admin settings are visible to non-admins (backend still checks for permission luckily)

**Comment by sean-brydon** [comment]
> related: #4270
> 
> it seems like all admin settings are visible to non-admins (backend still checks for permission luckily)

Should be fixed in update - will wait for @zomars blessing :)

**Comment by sean-brydon** [comment]
@zomars this creates a content flash if you are not an admin (if you manually navigate to an admin page) - not 100% sure the best way to prevent that do you have any ideas?

**Review by zomars — APPROVED** [review]

**Comment by zomars** [comment]
Not ideal but let's follow up instead

**Pull request merged**
